synapse/storage/search.py

```diff
@@ -399,3 +406,23 @@ def _to_postgres_options(options_dict):
     return "'%s'" % (
         ",".join("%s=%s" % (k, v) for k, v in options_dict.items()),
     )
+
+
+def _postgres_parse_query(search_term):
+    """Takes a plain unicode string from the user and converts it into a form
+    that can be passed to `to_tsquery(..)` postgres func. We use this so that
+    we can add prefix matching, which isn't something `plainto_tsquery` supports.
+    """
+    results = re.findall(r"([\w\-]+)", search_term, re.UNICODE)
```

Comment:
For the benefit of us who aren't familiar with regexeps it might be worth explaining what this code is doing.

Also since it is duplicated with the for sqlite below it might be worth lifting it out of these functions and moving it to the call site.